### PR TITLE
Update the concepts and how esc works pages

### PR DESCRIPTION
**Pull request by thoward**
This makes some changes to the ESC docs Concepts and How ESC Works pages. The goal of this is to reduce the overlap/redundancy between the two pages and structure them better. It also includes the "octopus" diagram from the ESC product homepage.

Fixes: #12861 #12863

### CODE REVIEW

**Comment by interurban** [comment]
Thx Troy adding @komalali and @arunkumar611 for review as well.

File: content/docs/esc/concepts/_index.md [hunk 1]
```diff
@@ -12,51 +12,60 @@ aliases:
   - /docs/concepts/environments/
 ---
 
-Do you have secrets and configuration that is copy/pasted around multiple environments, that is prone to drift and accidental disclosure? Have you ever made a change to a config or secret and were unsure what the impact would be? Is it hard for developers in your organization to get access to short-lived credentials to work in the environments they need to develop and deploy into? Do you struggle to audit access levels and who has accessed or changed your secrets and configuration?
+Pulumi ESC (Environments, Secrets, and Configuration) is a tool that simplifies how organizations manage secrets and configurations across multiple environments. It enables teams to compose collections of configuration and secrets, which can be consumed by various infrastructure and application services. ESC helps ensure security, consistency, and efficiency in handling secrets and configuration.
```
Comment (arunkumar611) [inline content/docs/esc/concepts/_index.md:15]:
```suggestion
Pulumi ESC (Environments, Secrets, and Configuration) simplifies how organizations manage secrets and configurations across multiple environments. It enables teams to compose collections of configuration and secrets called `environments`, which can be consumed by various infrastructure and application services. ESC helps ensure security, consistency, and efficiency in handling secrets and configuration.
```

File: content/docs/esc/concepts/_index.md [hunk 2]
```diff
@@ -12,51 +12,60 @@ aliases:
...
 
-Pulumi ESC (Environments, Secrets, and Configuration) enables teams to create collections of configuration and secrets called Environments. Teams can then access those environment collections using the `esc` CLI, `pulumi` CLI, Pulumi SDK, or Pulumi Cloud REST API for various application and infrastructure needs. These environments can be composed of other environments to allow teams increased flexibility and fine-grained access control. Teams can have as many environments as they need.
+Pulumi ESC is offered as a fully managed cloud service in [Pulumi Cloud](/docs/pulumi-cloud/). ESC is a standalone tool that can be applied to many uses cases. It has native integration with our other products, including Pulumi Infrastructure as Code (IaC). The [pulumi/esc project](https://github.com/pulumi/esc) is open source, and contains the evaluation engine for environments, the `esc` CLI.
```
Comment (arunkumar611) [inline content/docs/esc/concepts/_index.md:17]:
```suggestion
Pulumi ESC is offered both as a fully managed cloud service in [Pulumi Cloud](/docs/pulumi-cloud/) and self-hosted for scenarios that require isolated environments. ESC has native integration with several products and other Pulumi products, including Pulumi Infrastructure as Code (IaC). The [pulumi/esc project](https://github.com/pulumi/esc) is open source, and contains the evaluation engine for environments, the `esc` CLI.
```

File: content/docs/esc/concepts/_index.md [hunk 3]
```diff
@@ -12,51 +12,60 @@ aliases:
...
-Access permissions can be set to only allow select members or teams to "open" an environment and retrieve secrets.
+Pulumi ESC takes a distinct approach to managing secrets and configuration that is different from other secret managers. ESC emphasizes flexiblity and integration, and is specifically designed for managing secrets and configurations across complex multi-cloud environments. Whether used in conjunction with [Pulumi IaC](/docs/iac/) or as a standalone tool, ESC helps streamline operations, reduce duplication, and enhance security for teams across a wide range of use cases.
 
-For more detail, see [adding OIDC and Secrets providers](/docs/esc/environments/working-with-environments/#adding-oidc-and-secrets-providers).
+### Centralized management, composability, and reusability
 
-Please see the [providers list](/docs/esc/integrations/) for a full list of currently supported integrations and providers.
+Pulumi ESC aggregates secrets and configuration from different sources into *environments*. These environments can be composed from other environments, allowing for flexible organization and reuse without duplication.
 
-## Removing Duplication
```
Comment (arunkumar611) [inline content/docs/esc/concepts/_index.md:44]: I would keep this

File: content/docs/esc/concepts/how-esc-works.md [hunk 4]
```diff
@@ -11,48 +11,129 @@ menu:
...
+## Architecture
 
-![Pulumi ESC ecosystem](/docs/esc/assets/pulumi_esc.png)
+Pulumi ESC is a hosted service provided as part of [Pulumi Cloud](/docs/pulumi-cloud/). ESC stores your secrets and configuration and proxies access to other secret stores through provider plugins.
 
-The following details corresponds to the numbered sections in the above diagram:
+Using configuration-as-code, these secrets and configuration values are composed into *environments* which are defined in YAML. ESC then makes these environments available to targets via common mechanisms like environment variables, configuration files, and directly via the ESC API and multi-language SDKs.
 
-1. Pulumi ESC enables you to define environments, which contain collections of secrets and configuration. Each environment can be composed from multiple environments.
+{{< figure src="/docs/esc/assets/esc-octopus-diagram.png" caption="Figure: A diagram showing the architecture of Pulumi ESC.">}}
```
Comment (arunkumar611) [inline content/docs/esc/concepts/how-esc-works.md:22]: can you ensure we use the image consistency? In www.pulumi.com/esc - we use the other image for "How ESC works" - here we use this octopus diagram.

File: content/docs/esc/concepts/_index.md [hunk 5]
```diff
@@ -12,51 +12,60 @@ aliases:
...
-  salutation: hello
-  greeting: ${salutation}, ${name}
-```
+2. ***Sources***: ESC can *input* configuration and secrets from a variety of sources, and it has an extensible plugin model for integrating third-party sources. ESC's built-in support for dynamic secret providers, allows for best-practices like generating short-term credentials via OIDC, and dynamically pulling secret values at the time of use, for all major cloud providers.
 
-They can also contain complex [structured configuration](/docs/esc/environments/working-with-environments/#structured-configuration).
+3. ***Targets***: ESC *outputs* configuration and secrets to a variety of popular targets. Pulumi ESC has a rich API that allows for easy integration, and provides standard output mechanisms like environment variables and key files.
 
-## Dynamic Secret Providers
+4. ***Management***: ESC environments are centrally managed in Pulumi Cloud, and can be locked down with RBAC, versioned, tagged, and audited. ESC secrets are encrypted in flight and at rest.
```
Comment (arunkumar611) [inline content/docs/esc/concepts/_index.md:29]:
```suggestion
4. ***Management***: ESC environments are centrally managed in Pulumi Cloud, and can be permissioned with RBAC, versioned, tagged, and audited. ESC secrets are encrypted in flight and at rest.
```

File: content/docs/esc/concepts/_index.md [hunk 6]
```diff
@@ -12,51 +12,60 @@ aliases:
...
 
-Pulumi ESC supports referencing secret and configuration providers, which allow you to pull in secrets from OIDC connect providers for short lived credentials or from vaults/secret managers for all the major cloud providers.
+## The ESC Approach
 
-Access permissions can be set to only allow select members or teams to "open" an environment and retrieve secrets.
+Pulumi ESC takes a distinct approach to managing secrets and configuration that is different from other secret managers. ESC emphasizes flexiblity and integration, and is specifically designed for managing secrets and configurations across complex multi-cloud environments. Whether used in conjunction with [Pulumi IaC](/docs/iac/) or as a standalone tool, ESC helps streamline operations, reduce duplication, and enhance security for teams across a wide range of use cases.
 
-For more detail, see [adding OIDC and Secrets providers](/docs/esc/environments/working-with-environments/#adding-oidc-and-secrets-providers).
+### Centralized management, composability, and reusability
```
Comment (arunkumar611) [inline content/docs/esc/concepts/_index.md:35]: the explanation under this section seems to talk about only the "management" piece - more specifically RBAC - and doesn't talk anything about composability/reusability.... And composability and reusability are synonyms?

File: content/docs/esc/concepts/_index.md [hunk 7]
```diff
@@ -12,51 +12,60 @@ aliases:
...
 
-## Dynamic Secret Providers
+4. ***Management***: ESC environments are centrally managed in Pulumi Cloud, and can be locked down with RBAC, versioned, tagged, and audited. ESC secrets are encrypted in flight and at rest.
 
-Pulumi ESC supports referencing secret and configuration providers, which allow you to pull in secrets from OIDC connect providers for short lived credentials or from vaults/secret managers for all the major cloud providers.
+## The ESC Approach
 
-Access permissions can be set to only allow select members or teams to "open" an environment and retrieve secrets.
+Pulumi ESC takes a distinct approach to managing secrets and configuration that is different from other secret managers. ESC emphasizes flexiblity and integration, and is specifically designed for managing secrets and configurations across complex multi-cloud environments. Whether used in conjunction with [Pulumi IaC](/docs/iac/) or as a standalone tool, ESC helps streamline operations, reduce duplication, and enhance security for teams across a wide range of use cases.
```
Comment (arunkumar611) [inline content/docs/esc/concepts/_index.md:33]:
```suggestion
Pulumi ESC takes a distinct approach to managing secrets and configuration that is different from other secret managers. ESC emphasizes flexibility and an open-ecosystem approach to integrations and is specifically designed for managing secrets and configurations across complex multi-cloud environments. Whether used in conjunction with [Pulumi IaC](/docs/iac/) or as a standalone tool, ESC helps streamline operations, reduce duplication, and enhance security for teams across a wide range of use cases.
```

File: content/docs/esc/concepts/_index.md [hunk 8]
```diff
@@ -12,51 +12,60 @@ aliases:
...
 
-See below an example of how a customer might organize their environments:
+### Dynamic vs static configurations
 
-![A diagram showing how the different environments with team based organization](team_environments.png)
+Pulumi ESC supports both **static** configurations (e.g. simple key-value pairs) and **dynamic** configurations (values retrieved from third-party providers) in the same environment. This allows teams to mix-and-match the type of configuration they need.
+
+Pulumi ESC also supports **dynamic secret providers**, such as AWS OIDC, Azure KeyVault, GCP Secrets Manager, and more. This allows teams to pull short-lived credentials or other secrets dynamically from external sources.
+
+More detail on dynamic secret providers is available in [Adding OIDC and secrets providers](/docs/esc/environments/working-with-environments/#adding-oidc-and-secrets-providers). The [providers list](/docs/esc/integrations/) details the currently supported integrations.
```
Comment (arunkumar611) [inline content/docs/esc/concepts/_index.md:61]: this link seems incorrect

File: content/docs/esc/concepts/_index.md [hunk 9]
```diff
@@ -12,51 +12,60 @@ aliases:
...
+
+### Configuration-as-Code, automation, and integration everywhere
```
Comment (arunkumar611) [inline content/docs/esc/concepts/_index.md:63]: we should talk about composability here and not in the section above

File: content/docs/esc/concepts/_index.md [hunk 10]
```diff
@@ -12,51 +12,60 @@ aliases:
...
+
+Like our other products, Pulumi ESC uses an "as-code" approach to configuration and secrets. ESC environments can be composed, managed, and accessed using code written in TypeScript, JavaScript, Go, Python, or YAML. The `esc` CLI and our full-featured API allows for scripted use in automated environments like CI/CD. ESC is already deeply integrated into Pulumi IaC and Pulumi Cloud, and its plugin AP makes it easy to integrate ESC with any third-party product either as a provider or consumer.
```
[the remark this answers is not in the document]
Comment (arunkumar611) [reply content/docs/esc/concepts/_index.md:65]: What's Plugin AP?

File: content/docs/esc/concepts/how-esc-works.md [hunk 11]
```diff
@@ -11,48 +11,129 @@ menu:
...
+This article details how Pulumi ESC works, including its architecture, supported integrations, and core functionalities. To understand some of the core ESC concepts, like *environments*, *dynamic secrets*, and *configuration-as-code*, please read through the [ESC Concepts](/docs/esc/concepts/) doc.
 
-Pulumi ESC is offered as a fully managed cloud service in [Pulumi Cloud](/docs/pulumi-cloud/) (and Pulumi Cloud Self-hosted in the near future). The [pulumi/esc project](https://github.com/pulumi/esc) is open source, and contains the evaluation engine for environments, the esc CLI, and in the future, the extensible plugins for source and target integrations.
+## Architecture
 
-![Pulumi ESC ecosystem](/docs/esc/assets/pulumi_esc.png)
+Pulumi ESC is a hosted service provided as part of [Pulumi Cloud](/docs/pulumi-cloud/). ESC stores your secrets and configuration and proxies access to other secret stores through provider plugins.
```
Comment (arunkumar611) [inline content/docs/esc/concepts/how-esc-works.md:18]: again, add a note about self-hosted

File: content/docs/esc/concepts/how-esc-works.md [hunk 12]
```diff
@@ -11,48 +11,129 @@ menu:
...
 
-1. Pulumi ESC enables you to define environments, which contain collections of secrets and configuration. Each environment can be composed from multiple environments.
+{{< figure src="/docs/esc/assets/esc-octopus-diagram.png" caption="Figure: A diagram showing the architecture of Pulumi ESC.">}}
 
-2. Pulumi ESC supports a variety of configuration and secrets sources, and it has an extensible plugin model that allows third-party sources.
+## Secrets and configuration sources
 
-3. Pulumi ESC has a rich API that allows for easy integration. Every value in an environment can be accessed from any target execution environment.
+By default, Pulumi ESC stores your configuration and secrets in Pulumi Cloud. However, ESC also integrates with a variety of third-party sources through an extensible *provider* plugin model. This allows teams to use their preferred providers without needing to manually copy or paste secrets across environments. The secrets will be dynamically fetched from the third-party API and integrated into your ESC environments.
```
Comment (arunkumar611) [inline content/docs/esc/concepts/how-esc-works.md:26]: mention of providers here feels repetitive, we talked about it in the previous paragraph

File: content/docs/esc/concepts/how-esc-works.md [hunk 13]
```diff
@@ -11,48 +11,129 @@ menu:
...
 
-4. Every environment can be locked down with RBAC, versioned, and audited.
+By aggregating secrets from these providers, Pulumi ESC provides a unified interface for retrieving, using, and managing secrets, regardless of their source.
 
-## Dynamic Secrets Providers
+### Supported secret providers
```
Comment (arunkumar611) [inline content/docs/esc/concepts/how-esc-works.md:30]: I would call the "Dynamic Providers" and bucket them into Login providers and secret providers - just like the integrations left nav suggests

File: content/docs/esc/concepts/how-esc-works.md [hunk 14]
```diff
@@ -11,48 +11,129 @@ menu:
...
+### Configuration and Secrets as Code
 
-## Configuration as Code
+Pulumi ESC uses a configuration-as-code approach. ESC environments are defined in YAML, which contain named configuration values, encrypted secrets, and provider references.
 
-Environments are defined as YAML documents which can describe how to project and compose secrets and configuration, integrate dynamic configuration providers, and compute new configuration from other values (constructing a URL from a DNS name, or concatenating multiple configuration values into a derived value).  The incredible flexibility of a code-based approach over traditional point-and-click interfaces allows Pulumi ESC to offer rich expressiveness for managing complex configuration.
+A simple static environment can be thought of as a collection of key/value pairs. They can also contain interpolated values and complex [structured configuration](/docs/esc/environments/working-with-environments/#structured-configuration). Static secrets are also defined in YAML, but the static values are encrypted.
```
Comment (arunkumar611) [inline content/docs/esc/concepts/how-esc-works.md:48]:
```suggestion
A simple static environment can be thought of as a collection of key/value pairs. They can also contain interpolated values and complex [structured configuration](/docs/esc/environments/working-with-environments/#structured-configuration). Static secrets are also defined in YAML and are encrypted before they are stored to ensure security. 
```

File: content/docs/esc/concepts/how-esc-works.md [hunk 15]
```diff
@@ -11,48 +11,129 @@ menu:
...
+  structured_data:
+    active: true
+    nums:
+      - 1
+      - 2
 
-## Auditable
+  myPassword:
+    fn::secret:
+      ciphertext: ZXNjeAA....
```
Comment (arunkumar611) [inline content/docs/esc/concepts/how-esc-works.md:66]: ciphertext is the resulting output of a secret... I wonder whether we should specify the input and what happens after the use saves. Right now if they copy this text, it won't work

File: content/docs/esc/concepts/how-esc-works.md [hunk 16]
```diff
@@ -11,48 +11,129 @@ menu:
...
+**Example:** *Defining dynamic secrets and projecting environment variables.*
+
+```yaml
+values:
+  aws:
+    login:
+      fn::open::aws-login:
+        oidc:
+          roleArn: arn:aws:iam::01234567891011:role/some-role
+          sessionName: some-session
```
Comment (arunkumar611) [inline content/docs/esc/concepts/how-esc-works.md:82]:
```suggestion
          sessionName: some-session
          duration: 1h
```

**Comment by thoward** [comment]
I applied suggestions from feedback. There are still some higher level improvements that can be made. We can continue to refine in later iterations, but should get these changes published instead of letting them linger, then we can work from this basis for the next round of changes.

**Review by interurban — APPROVED** [review]

**Pull request merged**